src/main/perl/RuleBasedEditor.pm

```diff
@@ -0,0 +1,1175 @@
+    } elsif ($line_fmt == LINE_FORMAT_KEY_VAL_SET) {
+        $config_line = "set $keyword = $value";
+    } elsif ($line_fmt == LINE_FORMAT_KEY_VAL) {
+        $config_line = $keyword;
+        $config_line .= " $value" if $value;
+        # In trust (shift.conf) format, there should be only one blank between
+        # tokens and no trailing spaces.
+        $config_line =~ s/\s\s+/ /g;
+        $config_line =~ s/\s+$//;
+    } else {
+        *$self->{LOG}->error("$function_name: invalid line format ($line_fmt). Internal inconsistency.");
+    }
+
+    *$self->{LOG}->debug(2, "$function_name: Configuration line : >>$config_line<<");
+    return $config_line;
+}
+
+
+=pod
+
+=item _buildLinePattern
+
+This function builds a pattern that will match an existing configuration line for
+the configuration parameter specified. The pattern built takes into account the line format.
+Every whitespace in the pattern (configuration parameter) are replaced by \s+.
+If the line format is LINE_FORMAT_KEY_VAL, no whitespace is
+imposed at the end of the pattern, as these format can be used to write a configuration
+directive as a keyword with no value.
+
+Arguments :
+    config_param: parameter to update
+    line_fmt: line format (see LINE_FORMAT_xxx constants)
+    config_value: when defined, make it part of the pattern (used when multiple lines
+                  with the same keyword are allowed)
+
+Return value:
+    A string containing the pattern to use to match the line in the file or undef
+    in case of an internal error (missing argument).
+
+=cut
+
+sub _buildLinePattern
+{
+    my $function_name = "_buildLinePattern";
+    my ($self, $config_param, $line_fmt, $config_value) = @_;
+
+    unless ($config_param) {
+        *$self->{LOG}->error("$function_name: 'config_param' argument missing (internal error)");
+        return undef;
+    }
+    unless (defined($line_fmt)) {
+        *$self->{LOG}->error("$function_name: 'line_fmt' argument missing (internal error)");
+        return undef;
+    }
+    if (defined($config_value)) {
+        *$self->{LOG}->debug(2, "$function_name: configuration value '$config_value' will be added to the pattern");
+        $config_value =~ s/\\/\\\\/g;
+        $config_value =~ s/([\-\+\?\.\*\[\]()\^\$])/\\$1/g;
+        $config_value =~ s/\s+/\\s+/g;
+    } else {
+        $config_value = "";
+    }
+
+    # config_param is generally a keyword and in this case it contains no whitespace.
+    # A special case is when config_param (the rule keyword) is used to match a line
+    # without specifying a rule: in this case it may contains whitespaces. Remove strict
+    # matching of them (match any type/number of whitespaces at the same position).
+    # Look at %trust_config_rules in ncm-dpmlfc Perl module for an example.
+    $config_param =~ s/\s+/\\s+/g;
+
+    my $config_param_pattern;
+    if ($line_fmt == LINE_FORMAT_SH_VAR) {
+        $config_param_pattern = "#?\\s*$config_param=" . $config_value;
```

Comment:
same with appending the config_value. if all patterns append the value, do it at the end